src/noit_metric_director.c

```diff
@@ -461,48 +515,42 @@ noit_metric_director_register_search_on_thread(int thread_id, int64_t account_id
   account_search_t *searches = account_get_search(account_id, true);
   mtev_uuid_copy(as->check_uuid, check_uuid);
   as->ast = noit_metric_tag_search_ref(ast);
-  if(mtev_uuid_is_null(check_uuid)) {
-    unsigned long hash = CK_HS_HASH(&searches->search_asts[thread_id].hs, tag_search_registration_hash, as);
-    ck_spinlock_lock(&searches->search_asts[thread_id].lock);
-    ck_hs_put(&searches->search_asts[thread_id].hs, hash, as);
-    ck_spinlock_unlock(&searches->search_asts[thread_id].lock);
-  } else {
-    check_search_t *cs = check_get_search(searches, check_uuid, true);
-    unsigned long hash = CK_HS_HASH(&cs->search_asts[thread_id].hs, tag_search_registration_hash, as);
-    ck_spinlock_lock(&cs->search_asts[thread_id].lock);
-    ck_hs_put(&cs->search_asts[thread_id].hs, hash, as);
-    ck_spinlock_unlock(&cs->search_asts[thread_id].lock);
-  }
+
+  check_search_t *cs = check_get_search(searches, check_uuid, true);
```

Comment:
What is the reason that `check_uuid` cannot be null anymore?

Reply:
The null UUID is a misnomer -- it is actually a UUID with all zeros.  And now it is explcitly stored in the checks hash as such.

It is ever-so-slightly less efficient, but makes the structure much easier to understand.